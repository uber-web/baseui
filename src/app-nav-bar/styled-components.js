/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled, withStyle} from '../styles/index.js';
import {StyledListItem} from '../menu/index.js';

const StyledClearButton = styled<{$isFocusVisible: boolean}>(
  'button',
  ({$theme, $isFocusVisible}) => ({
    boxSizing: 'border-box',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
    color: $theme.colors.contentPrimary,
    borderLeftWidth: 0,
    borderTopWidth: 0,
    borderRightWidth: 0,
    borderBottomWidth: 0,
    paddingTop: '0',
    paddingBottom: '0',
    paddingLeft: '0',
    paddingRight: '0',
    marginLeft: 0,
    marginTop: 0,
    marginRight: 0,
    marginBottom: 0,
    outline: $isFocusVisible ? `3px solid ${$theme.colors.accent}` : 'none',
    outlineOffset: '-3px',
    WebkitAppearance: 'none',
    cursor: 'pointer',
  }),
);

export const StyledRoot = styled<{}>('div', props => {
  const {$theme} = props;
  return {
    ...$theme.typography.font300,
    boxSizing: 'border-box',
    backgroundColor: $theme.colors.backgroundPrimary,
    boxShadow: $theme.lighting.shadow600,
    width: '100%',
  };
});

export const StyledSpacing = styled<{}>('div', props => {
  const {$theme} = props;
  return {
    boxSizing: 'border-box',
    height: '100%',
    paddingTop: $theme.sizing.scale700,
    paddingBottom: $theme.sizing.scale700,
    display: 'flex',
    alignItems: 'center',
    [`@media screen and (max-width: ${$theme.breakpoints.medium - 1}px)`]: {
      paddingTop: $theme.sizing.scale400,
      paddingBottom: $theme.sizing.scale400,
    },
  };
});

export const StyledAppName = styled<{}>('span', ({$theme}) => ({
  ...$theme.typography.font650,
  color: $theme.colors.primary,
  textDecoration: 'none',
  [`@media screen and (max-width: ${$theme.breakpoints.medium - 1}px)`]: {
    ...$theme.typography.font550,
  },
}));

// $FlowFixMe
export const StyledSideMenuButton = withStyle<{}>(
  StyledClearButton,
  ({$theme}) => ({
    marginRight: $theme.sizing.scale600,
    paddingTop: $theme.sizing.scale100,
    paddingBottom: $theme.sizing.scale100,
    paddingLeft: $theme.sizing.scale100,
    paddingRight: $theme.sizing.scale100,
  }),
);

export const StyledPrimaryMenuContainer = styled<{}>('div', ({$theme}) => {
  return {
    boxSizing: 'border-box',
    height: '100%',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
    alignItems: 'stretch',
  };
});

export const StyledPrimaryMenuItem = styled<{
  $active: boolean,
  $isFocusVisible: boolean,
}>('div', props => {
  const {
    $active,
    $isFocusVisible,
    $theme: {colors, sizing},
  } = props;
  return {
    boxSizing: 'border-box',
    display: 'flex',
    alignItems: 'center',
    color: $active ? colors.contentPrimary : colors.contentTertiary,
    marginLeft: sizing.scale700,
    marginRight: sizing.scale700,
    outline: $isFocusVisible ? `3px solid ${colors.accent}` : 'none',
    outlineOffset: '-3px',
    borderBottom:
      $active && !$isFocusVisible
        ? `2px solid ${colors.primary}`
        : '2px solid transparent',
    ':hover': {
      color: colors.primary,
    },
  };
});

export const StyledUserMenuButton = StyledClearButton;

export const StyledUserMenuListItem = withStyle(StyledListItem, {
  paddingTop: '0',
  paddingBottom: '0',
  paddingRight: '0',
});

export const StyledUserProfileTileContainer = styled<{}>('div', ({$theme}) => {
  return {
    boxSizing: 'border-box',
    height: '100%',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'flex-start',
    paddingTop: $theme.sizing.scale650,
    paddingBottom: $theme.sizing.scale650,
  };
});

export const StyledUserProfilePictureContainer = styled<{}>(
  'div',
  ({$theme}) => {
    return {
      marginRight: $theme.sizing.scale600,
    };
  },
);

export const StyledUserProfileInfoContainer = styled<{}>('div', ({$theme}) => {
  return {
    boxSizing: 'border-box',
    alignSelf: 'center',
  };
});
