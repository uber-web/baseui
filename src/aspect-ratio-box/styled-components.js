/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';

export const Root = styled<{$aspectRatio: number}>('div', ({$aspectRatio}) => ({
  width: '100%',
  position: 'relative',
  '::before': {
    content: '""',
    width: '1px',
    marginLeft: '-1px',
    float: 'left',
    height: 0,
    paddingTop: `${100 / $aspectRatio}%`,
    pointerEvents: 'none',
  },
  '::after': {
    content: '""',
    display: 'table',
    clear: 'both',
  },
}));

export const Body = styled<{}>('div', () => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  width: '100%',
}));
