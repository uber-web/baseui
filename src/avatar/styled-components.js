/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';
import type {StylePropsT} from './types.js';

function getSize(props: StylePropsT) {
  const {$size, $theme} = props;

  const defaultSize = $theme.sizing.scale1000;
  const size = $size || defaultSize;
  return $theme.sizing[size] || size;
}

export const Avatar = styled('img', (props: StylePropsT) => {
  const themedSize = getSize(props);

  return {
    borderTopRightRadius: '50%',
    borderBottomRightRadius: '50%',
    borderTopLeftRadius: '50%',
    borderBottomLeftRadius: '50%',
    boxSizing: 'border-box',
    display: 'block',
    height: themedSize,
    width: themedSize,
    objectFit: 'cover',
  };
});

export const Initials = styled('div', (props: StylePropsT) => ({
  ...props.$theme.typography.font400,
  color: props.$theme.colors.mono100,
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  height: '100%',
}));

export const Root = styled('div', (props: StylePropsT) => {
  const {$didImageFailToLoad} = props;
  const themedSize = getSize(props);

  return {
    backgroundColor: $didImageFailToLoad ? props.$theme.colors.primary : null,
    borderTopRightRadius: '50%',
    borderBottomRightRadius: '50%',
    borderTopLeftRadius: '50%',
    borderBottomLeftRadius: '50%',
    boxSizing: 'border-box',
    display: 'inline-block',

    // image previously set the root height/width
    // since image is not rendered, set the height/width
    height: $didImageFailToLoad ? themedSize : null,
    width: $didImageFailToLoad ? themedSize : null,
  };
});
