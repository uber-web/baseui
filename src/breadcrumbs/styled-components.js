/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index.js';

export const StyledRoot = styled<{}>('nav', ({$theme}) => {
  return {
    color: $theme.colors.breadcrumbsText,
    ...$theme.typography.font450,
  };
});

export const StyledSeparator = styled<{}>('div', ({$theme}) => {
  return {
    display: 'inline-block',
    color: $theme.colors.breadcrumbsSeparatorFill,
    marginLeft: $theme.sizing.scale300,
    marginRight: $theme.sizing.scale300,
  };
});
