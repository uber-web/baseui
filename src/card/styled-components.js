/* @flow */

import {styled} from '../styles/index';

export const Action = styled('div', () => ({}));

export const Body = styled('div', ({$theme}) => {
  const {colors, sizing, typography} = $theme;
  return {
    marginBottom: sizing.scale600,
    color: colors.mono800,
    ...typography.font400,
  };
});

export const Contents = styled('div', ({$theme}) => {
  const {sizing} = $theme;
  return {
    margin: sizing.scale600,
  };
});

export const HeaderImage = styled('img', ({$theme}) => {
  const {sizing} = $theme;
  return {
    borderTopLeftRadius: sizing.scale100,
    borderTopRightRadius: sizing.scale100,
    objectFit: 'contain',
    maxWidth: '100%',
  };
});

export const Thumbnail = styled('img', props => {
  const {
    $theme: {border, sizing},
  } = props;
  return {
    float: 'right',
    height: sizing.scale2400,
    width: sizing.scale2400,
    objectFit: 'cover',
    borderRadius: sizing.scale100,
    ...border.border200,
    margin: `0 0 ${sizing.scale500} ${sizing.scale500}`,
  };
});

export const Title = styled('h1', ({$hasThumbnail, $theme}) => {
  const {colors, sizing, typography} = $theme;
  return {
    ...typography.font600,
    color: colors.black,
    fontSize: $hasThumbnail ? sizing.scale600 : sizing.scale700,
    fontWeight: 500,
    margin: `0 0 ${sizing.scale300} 0`,
    padding: 0,
  };
});

export const Wrapper = styled('div', ({$theme}) => {
  const {border, lighting, sizing} = $theme;
  return {
    ...border.border300,
    boxShadow: lighting.shadow400,
    borderRadius: sizing.scale100,
  };
});
