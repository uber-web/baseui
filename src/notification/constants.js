/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const KIND = Object.freeze({
  primary: 'primary',
  success: 'success',
  warning: 'warning',
  error: 'error',
});
