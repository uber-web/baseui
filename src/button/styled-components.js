/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {styled} from '../styles';
import {KIND, SIZE, SHAPE} from './constants';
import type {ThemeT} from '../styles';

type StylePropsT = {
  $theme: ThemeT,
  $size?: $Keys<typeof SIZE>,
  $kind?: $Keys<typeof KIND>,
  $shape?: $Keys<typeof SHAPE>,
  $isLoading?: boolean,
};

export const BaseButton = styled(
  'button',
  ({$theme, $size, $kind, $shape, $isLoading}: StylePropsT) => ({
    ...($size === SIZE.compact
      ? $theme.typography.font200
      : $theme.typography.font300),
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    border: 'none',
    borderRadius: $shape === SHAPE.round ? '50%' : $theme.borders.radius200,
    textDecoration: 'none',
    outline: 'none',
    WebkitAppearance: 'none',
    transitionProperty: 'background',
    transitionDuration: $theme.animation.timing100,
    transitionTimingFunction: $theme.animation.easeOutCurve,
    cursor: 'pointer',
    ':disabled': {
      cursor: 'not-allowed',
      backgroundColor: $theme.colors.mono300,
      color: $theme.colors.mono600,
    },
    // Padding For Shape and Size
    ...getStyleForShape({$theme, $shape, $size}),
    // Kind style override
    ...getStyleForKind({$theme, $kind, $isLoading}),
  }),
);

export const EndEnhancer = styled('div', ({$theme}: StylePropsT) => ({
  display: 'flex',
  marginLeft: $theme.sizing.scale500,
}));

export const StartEnhancer = styled('div', ({$theme}: StylePropsT) => ({
  display: 'flex',
  marginRight: $theme.sizing.scale500,
}));

export const LoadingSpinner = styled('div', ({$theme, $kind}: StylePropsT) => ({
  height: $theme.sizing.scale600,
  width: $theme.sizing.scale600,
  borderRadius: '50%',
  borderStyle: 'solid',
  borderWidth: $theme.sizing.scale0,
  borderTopColor:
    $kind === KIND.primary ? $theme.colors.white : $theme.colors.primary,
  borderLeftColor:
    $kind === KIND.primary
      ? 'rgba(255, 255, 255, 0.32)'
      : 'rgba(39, 110, 241, 0.32)',
  borderBottomColor:
    $kind === KIND.primary
      ? 'rgba(255, 255, 255, 0.32)'
      : 'rgba(39, 110, 241, 0.32)',
  borderRightColor:
    $kind === KIND.primary
      ? 'rgba(255, 255, 255, 0.32)'
      : 'rgba(39, 110, 241, 0.32)',
  animationDuration: $theme.animation.timing700,
  animationTimingFunction: 'linear',
  animationIterationCount: 'infinite',
  animationName: {
    to: {
      transform: 'rotate(360deg)',
    },
    from: {
      transform: 'rotate(0deg)',
    },
  },
}));

export function getStyleForShape({$theme, $shape, $size}: StylePropsT) {
  switch ($shape) {
    case SHAPE.round:
    case SHAPE.square:
      return {
        paddingTop:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
        paddingBottom:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
        paddingLeft:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
        paddingRight:
          $size === SIZE.compact
            ? $theme.sizing.scale400
            : $theme.sizing.scale500,
      };
    default:
      return {
        paddingTop:
          $size === SIZE.compact
            ? $theme.sizing.scale200
            : $theme.sizing.scale300,
        paddingBottom:
          $size === SIZE.compact
            ? $theme.sizing.scale200
            : $theme.sizing.scale300,
        paddingLeft: $theme.sizing.scale600,
        paddingRight: $theme.sizing.scale600,
      };
  }
}

export function getStyleForKind({$theme, $kind, $isLoading}: StylePropsT) {
  switch ($kind) {
    case KIND.primary:
      return {
        color: $theme.colors.white,
        backgroundColor: $theme.colors.primary,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.primary600
            : $theme.colors.primary500,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.primary600
            : $theme.colors.primary500,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.primary600,
        },
      };
    case KIND.secondary:
      return {
        color: $theme.colors.primary,
        backgroundColor: $theme.colors.primary50,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.primary200
            : $theme.colors.primary100,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.primary200
            : $theme.colors.primary100,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.primary200,
        },
      };
    case KIND.tertiary:
      return {
        color: $theme.colors.primary,
        backgroundColor: $theme.colors.mono200,
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.mono500
            : $theme.colors.mono400,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.mono500
            : $theme.colors.mono400,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.mono500,
        },
      };
    case KIND.minimal:
      return {
        color: $theme.colors.primary,
        backgroundColor: 'transparent',
        ':hover:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.mono400
            : $theme.colors.mono200,
        },
        ':focus:enabled': {
          backgroundColor: $isLoading
            ? $theme.colors.mono400
            : $theme.colors.mono200,
        },
        ':active:enabled': {
          backgroundColor: $theme.colors.mono400,
        },
      };
    default:
      return {};
  }
}
