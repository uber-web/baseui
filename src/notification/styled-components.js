/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index';
import type {StyledRootPropsT} from './types';
import {KIND} from './constants';

export const StyledRoot = styled('div', ({$theme, $kind}: StyledRootPropsT) => {
  let color = $theme.colors.notificationPrimaryText;
  let backgroundColor = $theme.colors.notificationPrimaryBackground;

  if ($kind === KIND.success) {
    color = $theme.colors.notificationSuccessText;
    backgroundColor = $theme.colors.notificationSuccessBackground;
  } else if ($kind === KIND.warning) {
    color = $theme.colors.notificationWarningText;
    backgroundColor = $theme.colors.notificationWarningBackground;
  } else if ($kind === KIND.error) {
    color = $theme.colors.notificationErrorText;
    backgroundColor = $theme.colors.notificationErrorBackground;
  }

  return {
    ...$theme.typography.font350,
    borderRadius: $theme.borders.radius200,
    paddingTop: $theme.sizing.scale600,
    paddingRight: $theme.sizing.scale600,
    paddingBottom: $theme.sizing.scale600,
    paddingLeft: $theme.sizing.scale600,
    color,
    backgroundColor,
  };
});
