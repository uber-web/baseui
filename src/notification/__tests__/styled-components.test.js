/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledRoot} from '../styled-components';
import {KIND} from '../constants';

describe('StyledComponents', () => {
  describe('StyledRoot', () => {
    it.each([[KIND.primary], [KIND.warning], [KIND.success], [KIND.error]])(
      'displays correct styles for %s notification',
      kind => {
        const component = shallow(<StyledRoot $kind={kind} />);
        expect(component.instance().getStyles()).toMatchSnapshot();
      },
    );
  });
});
