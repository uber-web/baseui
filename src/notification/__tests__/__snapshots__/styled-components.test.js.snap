// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledComponents StyledRoot displays correct styles for error notification 1`] = `
Object {
  "backgroundColor": "$theme.colors.notificationErrorBackground",
  "borderRadius": "$theme.borders.radius200",
  "color": "$theme.colors.notificationErrorText",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`StyledComponents StyledRoot displays correct styles for primary notification 1`] = `
Object {
  "backgroundColor": "$theme.colors.notificationPrimaryBackground",
  "borderRadius": "$theme.borders.radius200",
  "color": "$theme.colors.notificationPrimaryText",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`StyledComponents StyledRoot displays correct styles for success notification 1`] = `
Object {
  "backgroundColor": "$theme.colors.notificationSuccessBackground",
  "borderRadius": "$theme.borders.radius200",
  "color": "$theme.colors.notificationSuccessText",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;

exports[`StyledComponents StyledRoot displays correct styles for warning notification 1`] = `
Object {
  "backgroundColor": "$theme.colors.notificationWarningBackground",
  "borderRadius": "$theme.borders.radius200",
  "color": "$theme.colors.notificationWarningText",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": "$theme.typography.font350.fontWeight",
  "lineHeight": "$theme.typography.font350.lineHeight",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale600",
}
`;
